File: dart-language/charge-system-demo/client-ng-dart/lib/src/charge/charge_component.dart
import 'dart:js';
import 'package:angular/angular.dart';
import '../charge_service.dart';
import '../history/history_item_component.dart';

@Component(
    selector: 'my-charge',
    styleUrls: ['charge_component.css'],
    templateUrl: 'charge_component.html',
    directives: [coreDirectives, HistoryItemComponent])
class ChargeComponent implements OnInit {
  final ChargeService chargeService;
  List<History> histories = [];
  List<Choice> choices = [];

  ChargeComponent(this.chargeService);

  void ngOnInit() {
    this.getChoices();
    this.getHistoryBy4();
  }

  void confirmCharge(num value) async {
    final res =
        context.callMethod('confirm', ['Do you want to charge \$ $value?']);
    if (res) {
      final ok = await chargeService.postCharge(value);
      if (ok) {
        getHistoryBy4();
      }
    }
  }

  void getHistoryBy4() async {
    final res = await chargeService.getHistory();
    histories = res.sublist(0, 4);
  }

  void getChoices() async {
    final res = await chargeService.getChoices();
    choices = res;
  }
}

File: dart-language/charge-system-demo/client-ng-ts/projects/at-antd/src/app/charge/charge.component.ts
import { Component, OnInit } from '@angular/core';
import { Choice, History, ChargeService } from '../charge.service';
import { NzMessageService, NzModalService } from 'ng-zorro-antd';

@Component({
  selector: 'ant-charge',
  templateUrl: './charge.component.html',
  styleUrls: ['./charge.component.less']
})
export class ChargeComponent implements OnInit {
  choices: Choice[] = [];
  histories: History[] = [];

  constructor(
    private chargeService: ChargeService,
    private modalService: NzModalService,
    private message: NzMessageService
  ) { }

  ngOnInit() {
    this.getChoices();
    this.getRecentHistoryBy4();
  }
  getChoices() {
    this.chargeService.getChoices().subscribe(res => this.choices = res);
  }

  confirmCharge(value: number) {
    this.modalService.confirm({
      nzTitle: `<i>Do you want charge $ ${value}?</i>`,
      nzOnOk: () => { this.charge(value); }
    });
  }
  charge(value: number) {
    this.chargeService.postCharge(value).subscribe(res => {
      if (res != null) {
        this.message.create('success','Charged successfully!');
        this.getRecentHistoryBy4();
      }
    });

  }
  getRecentHistoryBy4() {
    this.chargeService.getHistory().subscribe(res => {
      if (res !== null) {
        this.histories = res.splice(0, 4);
      }
    });

  }

}
